src/components/MultiGallery/Gallery.tsx

```diff
@@ -239,78 +251,94 @@ export default function Gallery({ isFeatured = false, galleryRef, queryRef }: Pr
+            <HStack shrink justify="space-between">
+              <StyledGalleryTitleWrapper isHidden={hidden}>
+                <StyledIconContainer
+                  isDragging={isDragging}
+                  size="md"
+                  variant="stacked"
+                  icon={<DragHandleIconOutline color={colors.offBlack} />}
+                />
+                <VStack>
+                  <UnstyledLink href={galleryLink}>
+                    <StyledGalleryTitle tabIndex={1}>{name || 'Untitled'}</StyledGalleryTitle>
+                  </UnstyledLink>
+                  <BaseM>
+                    {collections.length} collection{collections.length === 1 ? '' : 's'}
+                  </BaseM>
+                </VStack>
+              </StyledGalleryTitleWrapper>
             </HStack>
-          </StyledGalleryActionsContainer>
-        </HStack>
-
-        <StyledTokenPreviewWrapper isHidden={hidden}>
-          {nonNullTokenPreviews.map((token) => (
-            <StyledTokenPreview key={token} src={token} />
-          ))}
-          {[...Array(remainingTokenPreviews).keys()].map((index) => (
-            <StyledEmptyTokenPreview key={index} />
-          ))}
-        </StyledTokenPreviewWrapper>
-      </StyledGalleryDraggable>
-    </StyledGalleryWrapper>
-  );
+            <StyledGalleryActionsContainer>
+              <HStack gap={8} align="center">
+                {isFeatured && (
+                  <StyledGalleryFeaturedText as="span">Featured</StyledGalleryFeaturedText>
+                )}
+                {isAuthenticatedUser && (
+                  <>
+                    <Link href={handleEditGallery}>
+                      <a>
+                        <IconContainer size="md" variant="stacked" icon={<EditPencilIcon />} />
+                      </a>
+                    </Link>
+                    <SettingsDropdown iconVariant="stacked">
+                      <DropdownSection>
+                        <DropdownItem onClick={(e) => handleEditGalleryName(e)}>
+                          EDIT NAME & DESC
+                        </DropdownItem>
+                        {hidden ? (
+                          <DropdownItem onClick={(e) => handleUpdateGalleryHidden(e)}>
+                            UNHIDE
+                          </DropdownItem>
+                        ) : (
+                          <>
+                            {!isFeatured && (
+                              <DropdownItem onClick={(e) => handleSetFeaturedGallery(e)}>
+                                FEATURE ON PROFILE
+                              </DropdownItem>
+                            )}
+                            <DropdownItem onClick={(e) => handleUpdateGalleryHidden(e)}>
+                              HIDE
+                            </DropdownItem>
+                          </>
+                        )}
+                        <DropdownItem onClick={(e) => handleDeleteGallery(e)}>DELETE</DropdownItem>
```

Comment:
@tbezman @Robinnnnn need some advice on what is the best way to handle this event bubbling. 

Current implementation works but felt weird. lol